pykeepass/kdbx_parsing/common.py

```diff
@@ -127,7 +127,13 @@ def compute_key_composite(password=None, keyfile=None):
         password_composite = b''
     # hash the keyfile
     if keyfile:
-        if hasattr(keyfile, "read"):
+        if (
+            isinstance(keyfile, io.BufferedIOBase)
+            or isinstance(keyfile, io.TextIOBase)
+            or isinstance(keyfile, io.RawIOBase)
```

Comment:
Is this a general way to check if an object is seekable?  Could we possibly just check `hasattr('seekable')`

Reply:
I used https://stackoverflow.com/questions/1661262/check-if-object-is-file-like-in-python as a reference. I am not sure if thats the best way to look at it tough.

Reply:
Actually you can use hasattr here. So let's give it a try.

Reply:
While you can, from the link above I have to wonder if it is the most sensible thing